### ISSUE TITLE
0.31.3 Server initialization failed

### ISSUE THREAD

**Issue by zero51** [castwide/solargraph#171]
macOS: 10.14.3
Ruby: 2.5.1
Solargraph: 0.31.3
VS Code: 1.32.3

Solargraph fails to startup:
```
[Error - 11:22:50 am] Server initialization failed.
  Message: [NoMethodError] undefined method `node' for #<Solargraph::Pin::Attribute:0x00007fa0869548f8>
  Code: -32603 
```

Stack traces:
```
console.ts:134 [Extension Host] Solargraph is listening PORT=49754 PID=16091

console.ts:134 [Extension Host] warning: parser/current is loading parser/ruby25, which recognizes
warning: 2.5.5-compliant syntax, but you are running 2.5.1.
warning: please see https://github.com/whitequark/parser#compatibility-with-ruby-mri.

console.ts:134 [Extension Host] [WARN] Error processing request: [NoMethodError] undefined method `node' for #<Solargraph::Pin::Attribute:0x00007f9c3fc12c98>

console.ts:134 [Extension Host] [WARN] ["/Users/dave/.asdf/installs/ruby/2.5.1/lib/ruby/gems/2.5.0/gems/solargraph-0.31.3/lib/solargraph/source_map/node_processor/send_node.rb:151:in `process_private_class_method'", "/Users/dave/.asdf/installs/ruby/2.5.1/lib/ruby/gems/2.5.0/gems/solargraph-0.31.3/lib/solargraph/source_map/node_processor/send_node.rb:38:in `process'", "/Users/dave/.asdf/installs/ruby/2.5.1/lib/ruby/gems/2.5.0/gems/solargraph-0.31.3/lib/solargraph/source_map/node_processor.rb:73:in `process'", "/Users/dave/.asdf/installs/ruby/2.5.1/lib/ruby/gems/2.5.0/gems/solargraph-0.31.3/lib/solargraph/source_map/node_processor/base.rb:41:in `block in process_children'", "/Users/dave/.asdf/installs/ruby/2.5.1/lib/ruby/gems/2.5.0/gems/solargraph-0.31.3/lib/solargraph/source_map/node_processor/base.rb:39:in `each'", "/Users/dave/.asdf/installs/ruby/2.5.1/lib/ruby/gems/2.5.0/gems/solargraph-0.31.3/lib/solargraph/source_map/node_processor/base.rb:39:in `process_children'", "/Users/dave/.asdf/installs/ruby/2.5.1/lib/ruby/gems/2.5.0/gems/solargraph-0.31.3/lib/solargraph/source_map/node_processor/begin_node.rb:6:in `process'", "/Users/dave/.asdf/installs/ruby/2.5.1/lib/ruby/gems/2.5.0/gems/solargraph-0.31.3/lib/solargraph/source_map/node_processor.rb:73:in `process'", "/Users/dave/.asdf/installs/ruby/2.5.1/lib/ruby/gems/2.5.0/gems/solargraph-0.31.3/lib/solargraph/source_map/node_processor/base.rb:41:in `block in process_children'", "/Users/dave/.asdf/installs/ruby/2.5.1/lib/ruby/gems/2.5.0/gems/solargraph-0.31.3/lib/solargraph/source_map/node_processor/base.rb:39:in `each'", "/Users/dave/.asdf/installs/ruby/2.5.1/lib/ruby/gems/2.5.0/gems/solargraph-0.31.3/lib/solargraph/source_map/node_processor/base.rb:39:in `process_children'", "/Users/dave/.asdf/installs/ruby/2.5.1/lib/ruby/gems/2.5.0/gems/solargraph-0.31.3/lib/solargraph/source_map/node_processor/namespace_node.rb:21:in `process'", "/Users/dave/.asdf/installs/ruby/2.5.1/lib/ruby/gems/2.5.0/gems/solargraph-0.31.3/lib/solargraph/source_map/node_processor.rb:73:in `process'", "/Users/dave/.asdf/installs/ruby/2.5.1/lib/ruby/gems/2.5.0/gems/solargraph-0.31.3/lib/solargraph/source_map/node_processor/base.rb:41:in `block in process_children'", "/Users/dave/.asdf/installs/ruby/2.5.1/lib/ruby/gems/2.5.0/gems/solargraph-0.31.3/lib/solargraph/source_map/node_processor/base.rb:39:in `each'", "/Users/dave/.asdf/installs/ruby/2.5.1/lib/ruby/gems/2.5.0/gems/solargraph-0.31.3/lib/solargraph/source_map/node_processor/base.rb:39:in `process_children'", "/Users/dave/.asdf/installs/ruby/2.5.1/lib/ruby/gems/2.5.0/gems/solargraph-0.31.3/lib/solargraph/source_map/node_processor/namespace_node.rb:21:in `process'", "/Users/dave/.asdf/installs/ruby/2.5.1/lib/ruby/gems/2.5.0/gems/solargraph-0.31.3/lib/solargraph/source_map/node_processor.rb:73:in `process'", "/Users/dave/.asdf/installs/ruby/2.5.1/lib/ruby/gems/2.5.0/gems/solargraph-0.31.3/lib/solargraph/source_map/mapper.rb:24:in `map'", "/Users/dave/.asdf/installs/ruby/2.5.1/lib/ruby/gems/2.5.0/gems/solargraph-0.31.3/lib/solargraph/source_map/mapper.rb:42:in `map'", "/Users/dave/.asdf/installs/ruby/2.5.1/lib/ruby/gems/2.5.0/gems/solargraph-0.31.3/lib/solargraph/source_map.rb:140:in `map'", "/Users/dave/.asdf/installs/ruby/2.5.1/lib/ruby/gems/2.5.0/gems/solargraph-0.31.3/lib/solargraph/api_map.rb:86:in `block in catalog'", "/Users/dave/.asdf/installs/ruby/2.5.1/lib/ruby/gems/2.5.0/gems/solargraph-0.31.3/lib/solargraph/api_map.rb:68:in `each'", "/Users/dave/.asdf/installs/ruby/2.5.1/lib/ruby/gems/2.5.0/gems/solargraph-0.31.3/lib/solargraph/api_map.rb:68:in `catalog'", "/Users/dave/.asdf/installs/ruby/2.5.1/lib/ruby/gems/2.5.0/gems/solargraph-0.31.3/lib/solargraph/library.rb:18:in `initialize'", "/Users/dave/.asdf/installs/ruby/2.5.1/lib/ruby/gems/2.5.0/gems/solargraph-0.31.3/lib/solargraph/library.rb:368:in `new'", "/Users/dave/.asdf/installs/ruby/2.5.1/lib/ruby/gems/2.5.0/gems/solargraph-0.31.3/lib/solargraph/library.rb:368:in `load'", "/Users/dave/.asdf/installs/ruby/2.5.1/lib/ruby/gems/2.5.0/gems/solargraph-0.31.3/lib/solargraph/language_server/host.rb:255:in `prepare'", "/Users/dave/.asdf/installs/ruby/2.5.1/lib/ruby/gems/2.5.0/gems/solargraph-0.31.3/lib/solargraph/language_server/host.rb:272:in `block in prepare_folders'", "/Users/dave/.asdf/installs/ruby/2.5.1/lib/ruby/gems/2.5.0/gems/solargraph-0.31.3/lib/solargraph/language_server/host.rb:271:in `each'", "/Users/dave/.asdf/installs/ruby/2.5.1/lib/ruby/gems/2.5.0/gems/solargraph-0.31.3/lib/solargraph/language_server/host.rb:271:in `prepare_folders'", "/Users/dave/.asdf/installs/ruby/2.5.1/lib/ruby/gems/2.5.0/gems/solargraph-0.31.3/lib/solargraph/language_server/message/initialize.rb:11:in `block in process'", "/Users/dave/.asdf/installs/ruby/2.5.1/lib/ruby/2.5.0/benchmark.rb:293:in `measure'", "/Users/dave/.asdf/installs/ruby/2.5.1/lib/ruby/gems/2.5.0/gems/solargraph-0.31.3/lib/solargraph/language_server/message/initialize.rb:8:in `process'", "/Users/dave/.asdf/installs/ruby/2.5.1/lib/ruby/gems/2.5.0/gems/solargraph-0.31.3/lib/solargraph/language_server/host.rb:94:in `receive'", "/Users/dave/.asdf/installs/ruby/2.5.1/lib/ruby/gems/2.5.0/gems/solargraph-0.31.3/lib/solargraph/language_server/transport/adapter.rb:33:in `process'", "/Users/dave/.asdf/installs/ruby/2.5.1/lib/ruby/gems/2.5.0/gems/solargraph-0.31.3/lib/solargraph/language_server/transport/adapter.rb:14:in `block in opening'", "/Users/dave/.asdf/installs/ruby/2.5.1/lib/ruby/gems/2.5.0/gems/solargraph-0.31.3/lib/solargraph/language_server/transport/data_reader.rb:53:in `parse_message_from_buffer'", "/Users/dave/.asdf/installs/ruby/2.5.1/lib/ruby/gems/2.5.0/gems/solargraph-0.31.3/lib/solargraph/language_server/transport/data_reader.rb:31:in `block in receive'", "/Users/dave/.asdf/installs/ruby/2.5.1/lib/ruby/gems/2.5.0/gems/solargraph-0.31.3/lib/solargraph/language_server/transport/data_reader.rb:26:in `each_char'", "/Users/dave/.asdf/installs/ruby/2.5.1/lib/ruby/gems/2.5.0/gems/solargraph-0.31.3/lib/solargraph/language_server/transport/data_reader.rb:26:in `receive'", "/Users/dave/.asdf/installs/ruby/2.5.1/lib/ruby/gems/2.5.0/gems/solargraph-0.31.3/lib/solargraph/language_server/transport/adapter.rb:25:in `sending'", "/Users/dave/.asdf/installs/ruby/2.5.1/lib/ruby/gems/2.5.0/gems/backport-0.3.0/lib/backport/client.rb:48:in `sending'", "/Users/dave/.asdf/installs/ruby/2.5.1/lib/ruby/gems/2.5.0/gems/backport-0.3.0/lib/backport/server/tcpip.rb:25:in `block (2 levels) in tick'", "/Users/dave/.asdf/installs/ruby/2.5.1/lib/ruby/gems/2.5.0/gems/backport-0.3.0/lib/backport/server/tcpip.rb:19:in `each'", "/Users/dave/.asdf/installs/ruby/2.5.1/lib/ruby/gems/2.5.0/gems/backport-0.3.0/lib/backport/server/tcpip.rb:19:in `block in tick'", "/Users/dave/.asdf/installs/ruby/2.5.1/lib/ruby/gems/2.5.0/gems/backport-0.3.0/lib/backport/server/tcpip.rb:18:in `synchronize'", "/Users/dave/.asdf/installs/ruby/2.5.1/lib/ruby/gems/2.5.0/gems/backport-0.3.0/lib/backport/server/tcpip.rb:18:in `tick'", "/Users/dave/.asdf/installs/ruby/2.5.1/lib/ruby/gems/2.5.0/gems/backport-0.3.0/lib/backport/machine.rb:56:in `each'", "/Users/dave/.asdf/installs/ruby/2.5.1/lib/ruby/gems/2.5.0/gems/backport-0.3.0/lib/backport/machine.rb:56:in `tick'", "/Users/dave/.asdf/installs/ruby/2.5.1/lib/ruby/gems/2.5.0/gems/backport-0.3.0/lib/backport/machine.rb:64:in `run_server_thread'", "/Users/dave/.asdf/installs/ruby/2.5.1/lib/ruby/gems/2.5.0/gems/backport-0.3.0/lib/backport/machine.rb:19:in `run'", "/Users/dave/.asdf/installs/ruby/2.5.1/lib/ruby/gems/2.5.0/gems/backport-0.3.0/lib/backport.rb:51:in `run'", "/Users/dave/.asdf/installs/ruby/2.5.1/lib/ruby/gems/2.5.0/gems/solargraph-0.31.3/lib/solargraph/shell.rb:25:in `socket'", "/Users/dave/.asdf/installs/ruby/2.5.1/lib/ruby/gems/2.5.0/gems/thor-0.20.3/lib/thor/command.rb:27:in `run'", "/Users/dave/.asdf/installs/ruby/2.5.1/lib/ruby/gems/2.5.0/gems/thor-0.20.3/lib/thor/invocation.rb:126:in `invoke_command'", "/Users/dave/.asdf/installs/ruby/2.5.1/lib/ruby/gems/2.5.0/gems/thor-0.20.3/lib/thor.rb:387:in `dispatch'", "/Users/dave/.asdf/installs/ruby/2.5.1/lib/ruby/gems/2.5.0/gems/thor-0.20.3/lib/thor/base.rb:466:in `start'", "/Users/dave/.asdf/installs/ruby/2.5.1/lib/ruby/gems/2.5.0/gems/

console.ts:134 [Extension Host] solargraph-0.31.3/bin/solargraph:5:in `<top (required)>'", "/Users/dave/.asdf/installs/ruby/2.5.1/bin/solargraph:23:in `load'", "/Users/dave/.asdf/installs/ruby/2.5.1/bin/solargraph:23:in `<main>'"]

notificationsAlerts.ts:40 [NoMethodError] undefined method `node' for #<Solargraph::Pin::Attribute:0x00007f9c3fc12c98>
onDidNotificationChange @ notificationsAlerts.ts:40

console.ts:134 [Extension Host] Error starting Solargraph socket provider Error: [NoMethodError] undefined method `node' for #<Solargraph::Pin::Attribute:0x00007f9c3fc12c98>
	at new ResponseError (/Users/dave/.vscode/extensions/castwide.solargraph-0.19.3/node_modules/vscode-languageclient/node_modules/vscode-jsonrpc/lib/messages.js:46:28)
	at handleResponse (/Users/dave/.vscode/extensions/castwide.solargraph-0.19.3/node_modules/vscode-languageclient/node_modules/vscode-jsonrpc/lib/main.js:430:48)
	at processMessageQueue (/Users/dave/.vscode/extensions/castwide.solargraph-0.19.3/node_modules/vscode-languageclient/node_modules/vscode-jsonrpc/lib/main.js:258:17)
	at Immediate.<anonymous> (/Users/dave/.vscode/extensions/castwide.solargraph-0.19.3/node_modules/vscode-languageclient/node_modules/vscode-jsonrpc/lib/main.js:242:13)
	at runCallback (timers.js:696:18)
	at tryOnImmediate (timers.js:667:5)
	at processImmediate (timers.js:649:5)

notificationsAlerts.ts:40 Failed to start Solargraph: Error: [NoMethodError] undefined method `node' for #<Solargraph::Pin::Attribute:0x00007f9c3fc12c98>
onDidNotificationChange @ notificationsAlerts.ts:40

console.ts:134 [Extension Host] rejected promise not handled within 1 second: Error: [NoMethodError] undefined method `node' for #<Solargraph::Pin::Attribute:0x00007f9c3fc12c98>
t.log @ console.ts:134

console.ts:134 [Extension Host] stack trace: Error: [NoMethodError] undefined method `node' for #<Solargraph::Pin::Attribute:0x00007f9c3fc12c98>
	at new ResponseError (/Users/dave/.vscode/extensions/castwide.solargraph-0.19.3/node_modules/vscode-languageclient/node_modules/vscode-jsonrpc/lib/messages.js:46:28)
	at handleResponse (/Users/dave/.vscode/extensions/castwide.solargraph-0.19.3/node_modules/vscode-languageclient/node_modules/vscode-jsonrpc/lib/main.js:430:48)
	at processMessageQueue (/Users/dave/.vscode/extensions/castwide.solargraph-0.19.3/node_modules/vscode-languageclient/node_modules/vscode-jsonrpc/lib/main.js:258:17)
	at Immediate.<anonymous> (/Users/dave/.vscode/extensions/castwide.solargraph-0.19.3/node_modules/vscode-languageclient/node_modules/vscode-jsonrpc/lib/main.js:242:13)
	at runCallback (timers.js:696:18)
	at tryOnImmediate (timers.js:667:5)
	at processImmediate (timers.js:649:5)
t.log @ console.ts:134

mainThreadExtensionService.ts:49 [[object Object]][NoMethodError] undefined method `node' for #<Solargraph::Pin::Attribute:0x00007f9c3fc12c98>
$onExtensionRuntimeError @ mainThreadExtensionService.ts:49

mainThreadExtensionService.ts:50 Error: [NoMethodError] undefined method `node' for #<Solargraph::Pin::Attribute:0x00007f9c3fc12c98>
	at new ResponseError (/Users/dave/.vscode/extensions/castwide.solargraph-0.19.3/node_modules/vscode-languageclient/node_modules/vscode-jsonrpc/lib/messages.js:46)
	at handleResponse (/Users/dave/.vscode/extensions/castwide.solargraph-0.19.3/node_modules/vscode-languageclient/node_modules/vscode-jsonrpc/lib/main.js:430)
	at processMessageQueue (/Users/dave/.vscode/extensions/castwide.solargraph-0.19.3/node_modules/vscode-languageclient/node_modules/vscode-jsonrpc/lib/main.js:258)
	at Immediate.<anonymous> (/Users/dave/.vscode/extensions/castwide.solargraph-0.19.3/node_modules/vscode-languageclient/node_modules/vscode-jsonrpc/lib/main.js:242)
	at runCallback (timers.js:696)
	at tryOnImmediate (timers.js:667)
	at processImmediate (timers.js:649)
```

**Comment by castwide** [comment]
I was able to reproduce this error by applying `private_class_method` to an attribute instead of a method. Minimal example:

```ruby
module Foo
  class << self
    attr_reader :bar
  end
  private_class_method :bar
end
```

I should be able to fix it for the next release.

**Comment by castwide** [comment]
Fixed in v0.32.0.
